module1/lessons/ruby_object_model_poly.md

```diff
@@ -0,0 +1,220 @@
+
+```ruby
+pry(main)> BasicObject.included_modules
+# => []
+
+pry(main)> Object.included_modules
+# => [PP::ObjectMixin, Kernel]
+
+pry(main)> Animal.included_modules
+# => [AnimalBehavior, PP::ObjectMixin, Kernel]
+
+pry(main)> Dog.included_modules
+# => [AnimalBehavior, PP::ObjectMixin, Kernel]
+```
+
+From this information, we can deduce that `BasicObject` doesn't include any modules, `Object` includes `PP:ObjectMixin` and `Kernel` (you don't need to worry about what those are), and `Animal` includes `AnimalBehavior`.
+
+Our updated diagram:
+
+![Imgur](https://i.imgur.com/f4pszOG.png)
+
+<section class="note">
+
+### Note
+
+JavaScript does not have native support for mixins within its class system.
+
+</section>
+
+## Chart Exercise
+
+Break into small groups.
+
+Using `.class`, `.ancestors`, `.included_modules`, and `.superclass`, diagram the Object Model of these several commonly-used Ruby classes: Hash, Array, String, Integer, and Float.
+
+# The Lookup Chain and Polymorphism
+
+In programming, polymorphism lets us use the same method or property in different classes or objects. Each class or object can have its own special version of a method through a process called **method overriding**.
+
+<section class="note">
+
+### Note
+
+Another form of polymorphism is **method overloading** which allows you to create multiple versions of the same method, each with different parameters. Neither Ruby or JavaScript support method overloading.
+
+</section>
+
+We now have a mental model for how Ruby manages classes, instances, superclasses, and modules, but why does it matter? The biggest implication of the Object Model is the **Lookup Chain**. We know that we can store methods in several places (class, superclass, module), but what is the exact order that Ruby looks for things? If a method is defined in several places, which one will Ruby use?
+
+Understanding the Lookup Chain is crucial for mastering Polymorphism in Software Development. This chain determines the method resolution path when a method is called on an object and what should happen if that method is not found directly on the object.
+
+<section class="note">
+
+### Note
+
+In JavaScript, the **prototype chain** plays a similar role to Ruby's Lookup Chain, affecting how methods and properties are accessed and overridden.
+
+</section>
+
+## Lookup Chain Exercise
+
+Complete [this activity](https://github.com/turingschool-examples/lookup_chain_exercise). In the first part of the activity, you will map the Object Model for a `Chair` instance, just as we did above. Then you will alter the code to explore the order of the Lookup Chain.
```

Comment:
The name of the main branch in this activity repo is still set as "master". I remember at one point there was an effort to change all of these instances. Is there an easy way to allow me access to get that changed over?

Reply:
If we move this activity into the new Big-SE repo, then we can solve this one as well. Honestly, to me, I think if we're using that big repo for all lesson activites, this should be added to it.

Reply:
Done.